discovery-provider/src/queries/get_premium_track_signatures.py

```diff
@@ -0,0 +1,320 @@
+import concurrent.futures
+import logging
+from collections import defaultdict
+from typing import Dict, List, Set
+
+from eth_typing import ChecksumAddress
+from sqlalchemy.orm.session import Session
+from src.models.tracks.track import Track
+from src.models.users.user import User
+from src.premium_content.premium_content_access_checker import (
+    premium_content_access_checker,
+)
+from src.premium_content.signature import get_premium_content_signature
+from src.queries.get_associated_user_wallet import get_associated_user_wallet
+from src.utils import db_session, web3_provider
+from web3 import Web3
+
+logger = logging.getLogger(__name__)
+
+erc721_abi = '[{"constant":true,"inputs":[{"internalType":"address","name":"owner","type":"address"}],"name":"balanceOf","outputs":[{"internalType":"uint256","name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"}]'
+erc1155_abi = '[{"inputs":[{"internalType":"address","name":"account","type":"address"},{"internalType":"uint256","name":"id","type":"uint256"}],"name":"balanceOf","outputs":[{"internalType":"uint256","name":"","type":"uint256"}],"stateMutability":"view","type":"function"},{"inputs":[{"internalType":"address[]","name":"accounts","type":"address[]"},{"internalType":"uint256[]","name":"ids","type":"uint256[]"}],"name":"balanceOfBatch","outputs":[{"internalType":"uint256[]","name":"","type":"uint256[]"}],"stateMutability":"view","type":"function"}]'
+
+eth_web3 = web3_provider.get_eth_web3()
+
+
+def _get_user_wallet(user_id: int, session: Session):
```

Comment:
nitty nit: we should decide on a convention about whether we prefix private methods with `_`, we do this pretty inconsistently in DN codebase